### PR TITLE
Fix/Separate deploying contract and NNS updating transaction

### LINKED ISSUE #1675: Tuning default `MaxGasInvoke` RPC server setting of neofs devenv node
## Is your feature request related to a problem? Please describe.
The problem is related to the `neofs-adm morph deploy` command. Consider the situation of contract deploying (not updating):
```
$ neofs-adm morph deploy -r http://morph-chain.neofs.devenv:30333 --contract test --domain neofs --alphabet-wallets .
Password for az wallet > 
Contract group wallet is missing, initialize at contract.json
Password for contract wallet > 
NNS: Set license.neofs -> 651d138028e04a657de277d15e74934b4a1775b0
Error: can't create tx: can't add system fee to transaction: bad vm state: FAULT due to an error: at instruction 1944 (SYSCALL): failed to invoke syscall 3163314883: GAS limit exceeded
```
The described error happens during test invocation of contract deploying script and the reason is that amount of GAS allowed for test RPC invocation (which is 15, as described in the `MaxGasInvoke` node configuration) is exceeded. If we're to look at the deploying script, we can see that it includes:
1. The contract deploy itself: https://github.com/nspcc-dev/neofs-node/blob/cd71de69a08d87a63fedd7bc4ead54c4ddac8791/cmd/neofs-adm/internal/modules/morph/deploy.go#L107-L112
2. Corresponding NNS domain registration and new NNS record creation: https://github.com/nspcc-dev/neofs-node/blob/cd71de69a08d87a63fedd7bc4ead54c4ddac8791/cmd/neofs-adm/internal/modules/morph/deploy.go#L114

As a result, the script looks like the following:
```
NEO-GO-VM > loadbase64 wg2uAXsibmFtZSI6ImxpY2Vuc2UiLCJhYmkiOnsibWV0aG9kcyI6W3sibmFtZSI6Im1haW4iLCJvZmZzZXQiOjAsInBhcmFtZXRlcnMiOltdLCJyZXR1cm50eXBlIjoiVm9pZCIsInNhZmUiOmZhbHNlfV0sImV2ZW50cyI6W119LCJmZWF0dXJlcyI6e30sImdyb3VwcyI6W3sicHVia2V5IjoiMDM1YmM4ZjQyYjVkYjcyY2U1NzIwYTgwOThmN2M0YjQxOTBhZjNiNjFmYzU5OTlmZmEyZWNjN2MyMWJjZGU0ZWMyIiwic2lnbmF0dXJlIjoiNzhudlBoY0NmOWN4VHhlSUc2NmdiRE5aMzZyZGJ0NnpYdmY0SjRaVTZpQTJ5U3lqenliOURCdzhFcEo0c3JtTytxaUxOa1BNNVhBNEhGbDRuMGhRSHc9PSJ9XSwicGVybWlzc2lvbnMiOlt7ImNvbnRyYWN0IjoiKiIsIm1ldGhvZHMiOiIqIn1dLCJzdXBwb3J0ZWRzdGFuZGFyZHMiOltdLCJ0cnVzdHMiOltdLCJleHRyYSI6bnVsbH0MdE5FRjNuZW8tZ28tMC45OS4yLXByZS0yOS1nMjViZDk0MWQAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAEWh0dHA6Ly95YWRyby5jb20vAAAAABUMDUhlbGxvLCB3b3JsZCFBz+dHlkApgnzLE8AfDAZkZXBsb3kMFP2j+kNG6lMqJY/El92t22Q3yf3/QWJ9W1JFARAOAoA6CQABWAIBEA4MDG9wc0Buc3BjYy5ydQwU2MxakE1Sb8mmyrVboCyab+eJyVYMDWxpY2Vuc2UubmVvZnMXwB8MCHJlZ2lzdGVyDBRn1hVsZ+Kpbo94o4SOsna51DxsjkFifVtSOQwoNjUxZDEzODAyOGUwNGE2NTdkZTI3N2QxNWU3NDkzNGI0YTE3NzViMAAQDA1saWNlbnNlLm5lb2ZzE8AfDAlhZGRSZWNvcmQMFGfWFWxn4qluj3ijhI6ydrnUPGyOQWJ9W1I=
READY: loaded 797 instructions
NEO-GO-VM 0 > ops
INDEX    OPCODE       PARAMETER
0        NEWARRAY0        <<
1        PUSHDATA2    7b226e616d65223a226c6963656e7365222c22616269223a7b226d6574686f6473223a5b7b226e616d65223a226d61696e222c226f6666736574223a302c22706172616d6574657273223a5b5d2c2272657475726e74797065223a22566f6964222c2273616665223a66616c73657d5d2c226576656e7473223a5b5d7d2c226665617475726573223a7b7d2c2267726f757073223a5b7b227075626b6579223a22303335626338663432623564623732636535373230613830393866376334623431393061663362363166633539393966666132656363376332316263646534656332222c227369676e6174757265223a2237386e765068634366396378547865494736366762444e5a333672646274367a587666344a345a55366941327953796a7a7962394442773845704a3473726d4f2b71694c4e6b504d3558413448466c346e30685148773d3d227d5d2c227065726d697373696f6e73223a5b7b22636f6e7472616374223a222a222c226d6574686f6473223a222a227d5d2c22737570706f727465647374616e6461726473223a5b5d2c22747275737473223a5b5d2c226578747261223a6e756c6c7d ("{\"name\":\"license\",\"abi\":{\"methods\":[{\"name\":\"main\",\"offset\":0,\"parameters\":[],\"returntype\":\"Void\",\"safe\":false}],\"events\":[]},\"features\":{},\"groups\":[{\"pubkey\":\"035bc8f42b5db72ce5720a8098f7c4b4190af3b61fc5999ffa2ecc7c21bcde4ec2\",\"signature\":\"78nvPhcCf9cxTxeIG66gbDNZ36rdbt6zXvf4J4ZU6iA2ySyjzyb9DBw8EpJ4srmO+qiLNkPM5XA4HFl4n0hQHw==\"}],\"permissions\":[{\"contract\":\"*\",\"methods\":\"*\"}],\"supportedstandards\":[],\"trusts\":[],\"extra\":null}")
434      PUSHDATA1    4e4546336e656f2d676f2d302e39392e322d7072652d32392d6732356264393431640000000000000000000000000000000000000000000000000000000000000000000011687474703a2f2f796164726f2e636f6d2f00000000150c0d48656c6c6f2c20776f726c642141cfe747964029827ccb
552      PUSH3
553      PACK
554      PUSH15
555      PUSHDATA1    6465706c6f79 ("deploy")
563      PUSHDATA1    fda3fa4346ea532a258fc497ddaddb6437c9fdff
585      SYSCALL      System.Contract.Call (627d5b52)
590      DROP
591      PUSHINT16    3600 (100e)
594      PUSHINT32    604800 (803a0900)
599      PUSHINT16    600 (5802)
602      PUSHINT16    3600 (100e)
605      PUSHDATA1    6f7073406e737063632e7275 ("ops@nspcc.ru")
619      PUSHDATA1    d8cc5a904d526fc9a6cab55ba02c9a6fe789c956
641      PUSHDATA1    6c6963656e73652e6e656f6673 ("license.neofs")
656      PUSH7
657      PACK
658      PUSH15
659      PUSHDATA1    7265676973746572 ("register")
669      PUSHDATA1    67d6156c67e2a96e8f78a3848eb276b9d43c6c8e
691      SYSCALL      System.Contract.Call (627d5b52)
696      ASSERT
697      PUSHDATA1    36353164313338303238653034613635376465323737643135653734393334623461313737356230 ("651d138028e04a657de277d15e74934b4a1775b0")
739      PUSHINT8     16 (10)
741      PUSHDATA1    6c6963656e73652e6e656f6673 ("license.neofs")
756      PUSH3
757      PACK
758      PUSH15
759      PUSHDATA1    6164645265636f7264 ("addRecord")
770      PUSHDATA1    67d6156c67e2a96e8f78a3848eb276b9d43c6c8e
792      SYSCALL      System.Contract.Call (627d5b52)
```

The problem is that only new NNS domain registration costs more than 10 GAS, and the contract deployment (even for a tiny contract) is rather expensive too. As a result, the whole script costs more than 15 GAS which doesn't fit the RPC server limit (devenv node with default config running locally on http://morph-chain.neofs.devenv:30333).

## Describe the solution you'd like
The desired behaviour is to be able to deploy a contract with the help of  `$ neofs-adm morph deploy` command without making any additional changes in the devenv configuration.

I have two options as possible solution:
1. Increase `MaxGasInvoke` RPC server setting for the target devenv node. I've tested 50 GAS and the deployment passes successfully.
2. Decrease the cost of new NNS domain registration. Currently it is set to be 10 GAS, is it based on some calculations?

## Additional context
Also, I have a question. On new contract addition, the new NNS domain registration occurs according to the following rule: https://github.com/nspcc-dev/neofs-node/blob/cd71de69a08d87a63fedd7bc4ead54c4ddac8791/cmd/neofs-adm/internal/modules/morph/initialize_nns.go#L127-L137
E.g. we have the default `.neofs` NNS domain registered and want to deploy new `test` contract. As a result, new `test.neofs` NNS domain will be registered and `test.neofs -> 0x123...456`  NNS record will be added. The question is why do we have to register new domain each time the new contract is deployed? Isn't it enough to add  `test.neofs -> 0x123...456` NNS record?

### PR DESCRIPTION
**Pull request by carpawell**
Deploy contract in one transaction and register NNS record in another to
prevent GAS limit exceeding due to the fact that registration always takes
10+ GAS.

Signed-off-by: Pavel Karpy <carpawell@nspcc.ru>

Closes #1675.

### CODE REVIEW

**Comment by fyrchik** [comment]
NNS has `SetPrice` method, can we do `SetPrice` the same trick as in https://github.com/nspcc-dev/neofs-node/pull/1608 ?

**Comment by carpawell** [comment]
> NNS has SetPrice method, can we do SetPrice the same trick as in https://github.com/nspcc-dev/neofs-node/pull/1608 ?

I did like it was done for NeoFS contracts (deploying in [one stage](https://github.com/nspcc-dev/neofs-node/blob/89cba2cc7becdaadc55eb4d2996153318df5ddf6/cmd/neofs-adm/internal/modules/morph/initialize.go#L74), setting NNS in [another](https://github.com/nspcc-dev/neofs-node/blob/89cba2cc7becdaadc55eb4d2996153318df5ddf6/cmd/neofs-adm/internal/modules/morph/initialize.go#L94))

Also, if Im not mistaken, @realloc was against changing the price for that purpose.

**Comment by realloc** [comment]
I didn't like the idea of changing NNS pricing in general for all.

I have nothing against temporary NNS price change, contract deploy and setting price back in the same Tx. This doesn't mean we don't have to check deploy actions sanity first =)

**Comment by fyrchik** [comment]
Yes, the whole purpose of the optimisation is to do it temporarily, so that there are no visible side-effects for other transactions.

**Comment by carpawell** [comment]
@fyrchik, added hack with changing price but kept separate deploying and registering. Also, added `--force` that allows deploying contract without group/with a group that differs from `group.neofs`.

File: cmd/neofs-adm/internal/modules/morph/deploy.go [hunk 1]
```diff
@@ -109,24 +110,44 @@ func deployContractCmd(cmd *cobra.Command, args []string) error {
...
+	if err != nil {
+		return fmt.Errorf("can't wait deploying transaction: %w", err)
+	}
+
+	if isUpdate {
+		return nil
+	}
+
+	// add record to NNS
+	w = io.NewBufBinWriter()
```
Comment (AnnaShaleva) [inline cmd/neofs-adm/internal/modules/morph/deploy.go:155]: BufBinWriter can be reused after previous transaction. Use `w.Reset()` to reset it.

File: cmd/neofs-adm/internal/modules/morph/deploy.go [hunk 2]
```diff
@@ -95,9 +98,32 @@ func deployContractCmd(cmd *cobra.Command, args []string) error {
...
+		pub := c.ContractWallet.Accounts[0].PrivateKey().PublicKey()
+		group, err := nnsResolveKey(c.Client, nnsCs.Hash, morphclient.NNSGroupKeyName)
+		if err != nil {
+			return fmt.Errorf("can't resolve group key via NNS: %w", err)
+		}
+
+		if !pub.Equal(group) && !force {
+			return fmt.Errorf(
+				"`%s` does not equal NNS's `%s`; use `--%s` flag to deploy with that key anyway",
+				contractWalletFilename, morphclient.NNSGroupKeyName, forceFlag,
```
Comment (AnnaShaleva) [inline cmd/neofs-adm/internal/modules/morph/deploy.go:118]: Maybe some "%s doesn't contain NNS's group key..." will be better? The first argument is wallet filename, this message looks strange a bit.

File: cmd/neofs-adm/internal/modules/morph/deploy.go [hunk 3]
```diff
@@ -95,9 +98,32 @@ func deployContractCmd(cmd *cobra.Command, args []string) error {
...
+			)
+		}
+
+		err = c.addManifestGroup(cs.Hash, cs)
+		if err != nil {
+			return fmt.Errorf("can't sign manifest group: %v", err)
+		}
+
```
Comment (AnnaShaleva) [inline cmd/neofs-adm/internal/modules/morph/deploy.go:126]: Linter is unhappy.

Comment (carpawell) [reply cmd/neofs-adm/internal/modules/morph/deploy.go:126]: forget that the linter already works

File: cmd/neofs-adm/internal/modules/morph/deploy.go (hunk 2 above)
Comment (carpawell) [reply cmd/neofs-adm/internal/modules/morph/deploy.go:118]: decided not to deploy custom contracts with `group.neofs` group, removed that code

File: cmd/neofs-adm/internal/modules/morph/deploy.go (hunk 1 above)
Comment (carpawell) [reply cmd/neofs-adm/internal/modules/morph/deploy.go:155]: fixed

**Comment by carpawell** [comment]
Removed `--force` flag and do deploy contract with the neofs group.

**Comment by fyrchik** [comment]
@carpawell why did you separate deploy and register? `setPrice` should be enough.

**Comment by carpawell** [comment]
> why did you separate deploy and register?

@fyrchik that was the original idea of the PR. Yes, now `SetPrice` fixes the problem. Do you want to keep registration and deploying in the same TX?

**Comment by fyrchik** [comment]
Yes, the less transactions we have, the fewer places where something could go wrong.

**Comment by carpawell** [comment]
@fyrchik, ok, dropped that commit.

**Review by fyrchik — APPROVED** [review]

**Pull request merged**